internal/blocksync/reactor.go

```diff
@@ -250,12 +250,16 @@ func (r *Reactor) handleMessage(ctx context.Context, envelope *p2p.Envelope, blo
 					"err", err)
 				return err
 			}
-			extCommit, err := types.ExtendedCommitFromProto(msg.ExtCommit)
-			if err != nil {
-				r.logger.Error("failed to convert extended commit from proto",
-					"peer", envelope.From,
-					"err", err)
-				return err
+			var extCommit *types.ExtendedCommit
+			if msg.ExtCommit != nil {
```

Comment:
This is the point I asked yesterday (I think) whether we wanted to reject messages without extCommit, but when the `VoteExtensionsEnabled` is true.

You replied that we can't do this, as (since we're late) `VoteExtensionsEnabled` might not reflect the actual value.

I still agree with that. Just writing it here again to make sure we're still on the same page.

Reply:
Yes, this logic simply continues to save the block data even if `msg.ExtCommit` is nil. We cannot safely check if `msg.ExtCommit` should be non-nil until we process it in the `poolRoutine` loop.